models/comments.models.js

```diff
@@ -1,23 +1,25 @@
 const db = require('../db/connection')
-const articles = require('../db/data/test-data/articles')
-exports.selectCommentsByArticleId = (article_id) => {
-    let queryStr ='SELECT comments.comment_id, comments.votes, comments.created_at, comments.author, comments.body, comments.article_id FROM comments WHERE comments.article_id = $1'
-    let count = 0
-    const ValidIdArr = articles.map(()=>{
-        count++
-        return count
-    })
+// const{checkExists}=require('../utils.js')
+const{checkArticleIdExists, checkExists} =require('../utils')
 
-    const idIsValid = ValidIdArr.indexOf(Number(article_id))!==-1
+exports.selectCommentsByArticleId = (article_id) => {
+  const queryVals =[article_id]
+  const queryProms = []
+    let queryStr ='SELECT comments.comment_id, comments.votes, comments.created_at, comments.author, comments.body, comments.article_id FROM comments WHERE comments.article_id = $1 ORDER BY created_at'
 
-    queryStr += ` ORDER BY created_at;`
-    return db
-      .query(queryStr, [article_id])
-      .then(({rows}) => {
-        if(idIsValid){return rows}
-        if(rows.length===0){return Promise.reject({ message: "article does not exist", status: 404})
-        }
-    console.log(rows)
-        return rows;
+    queryProms.push(db.query(queryStr, queryVals))
+    queryProms.push(checkExists('articles','article_id', article_id))
+    return Promise.all(queryProms).then((output) => {
+        return output[0].rows;
       });
+}
+exports.insertComment = (article_id, {username, body}) =>{
+  const queryProms = []
+  let queryStr = 'INSERT INTO comments (body, author, article_id) VALUES ($1, $2, $3) RETURNING *;'
+  queryProms.push(db.query(queryStr, [body, username, article_id]))
+  queryProms.push(checkExists('articles','article_id', article_id))
```

Comment:
This step isn't actually necessary - the article id is a foreign key, so if the article id didn't exist, then psql would send you an error saying you've 'violated a foreign key constraint' which would make the comment insertion fail, without having to use checkExists